### PR TITLE
Replace clock() with std::chrono::high_resolution_clock()

### PR DESCRIPTION
**Pull request by rafie**

### CODE REVIEW

**Review by ashtul — DISMISSED** [review]
Does it have effect on performance?

File: src/aggregate/aggregate.h [hunk 1]
```diff
@@ -114,10 +115,10 @@ typedef struct {
   unsigned int dialectVersion;
 
   /** Profile variables */
-  clock_t initClock;          // Time of start. Reset for each cursor call
-  clock_t totalTime;          // Total time. Used to accimulate cursors times
-  clock_t parseTime;          // Time for parsing the query
-  clock_t pipelineBuildTime;  // Time for creating the pipeline
+  steady_clock_t initClock;  // Time of start. Reset for each cursor call
+  double totalTime;          // Total time. Used to accimulate cursors times
```
[the remark this answers is not in the document]
Comment (oshadmi) [reply src/aggregate/aggregate.h:119]: Add mention it is in micro sec?

**Review by oshadmi — COMMENTED** [review]
👍🏼 Just minor comments

**Review by oshadmi — APPROVED** [review]

**Review by ashtul — APPROVED** [review]

**Comment by MeirShpilraien** [comment]
Consider updating top comment and explain why we prefer to use this clock over what we already have today and if there are any concerns/things that should be taken into considerations (OS support/ new dependencies ...)

**Comment by rafie** [comment]
@MeirShpilraien Missed your remark before merging. I'll add the comments in a new PR.

**Pull request merged**